core/modules/cart/store/actions/itemActions.ts

```diff
@@ -80,9 +80,18 @@ const itemActions = {
         productIndex++
       }
     }
-    await dispatch('create')
+
+    let newDiffLog = await dispatch('create')
+    if (newDiffLog !== undefined) {
+      diffLog.merge(newDiffLog)
+    }
+
     if (getters.isCartSyncEnabled && getters.isCartConnected && !forceServerSilence) {
-      return dispatch('sync', { forceClientState: true })
+      await dispatch('sync', { forceClientState: true }).then((syncDiffLog) => {
```

Comment:
Can you use only async/await here? Lets not mix Promise with async/await. I think this should also work:
```
const syncDiffLog = await dispatch('sync', { forceClientState: true })
if (!syncDiffLog.isEmpty()) {
  diffLog.merge(syncDiffLog)
}
```